transport/spi/src/main/kotlin/Endpoint.kt

```diff
@@ -51,21 +51,34 @@ sealed class Endpoint<T : Any>(
     val configPrefix: String
 ) {
     companion object {
+        /**
+         * A map holding the existing [Endpoint] instances using their configuration prefixes as keys.
+         */
+        private val endpointsByConfigPrefix by lazy {
```

Comment:
Is there really anything expensive in creating this list, so that it's worth to create it only lazily?

Reply:
Without the `lazy`, an exception during class initialization is thrown. This is because the objects to be added to the map have not yet been fully initialized.